### PR TITLE
SimpleGazetteerAnnotator

### PR DESCRIPTION
**Pull request by cowchipkid**
Added the SimpleGazetteerAnnotator class which creates a SpanLabelView with all the gaz hits.

### CODE REVIEW

**Comment by cowchipkid** [comment]
For some reason, a check has failed on this pull request. A build failed on Semaphore?

**Comment by cowchipkid** [comment]
Unlike the implementation for NER, this gazetteer annotator is not a static singleton, meaning we could create many instances of the same Annotator over and over representing the same set of gazetteers which is very wasteful. I would prefer to have a factory method that produces unique annotator only if that set of gazetteers had not yet been loaded.

**Comment by mssammon** [comment]
@cowchipkid here is the error from semaphore:
[INFO] Reactor Summary:
[INFO] 
[INFO] illinois-cogcomp-nlp ............................... SUCCESS [  0.073 s]
[INFO] illinois-core-utilities ............................ SUCCESS [  5.808 s]
[INFO] illinois-tokenizer ................................. SUCCESS [  1.446 s]
[INFO] illinois-corpusreaders ............................. SUCCESS [  1.703 s]
[INFO] illinois-curator ................................... SUCCESS [  1.042 s]
[INFO] illinois-edison .................................... FAILURE [  1.301 s]
[INFO] illinois-lemmatizer ................................ SKIPPED
[INFO] illinois-pos ....................................... SKIPPED
[INFO] illinois-chunker ................................... SKIPPED
[INFO] illinois-ner ....................................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 11.748 s
[INFO] Finished at: 2016-05-20T16:17:01+00:00
[INFO] Final Memory: 76M/858M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:2.0.2:compile (default-compile) on project illinois-edison: Compilation failure: Compilation failure:
[ERROR] /home/runner/illinois-cogcomp-nlp/edison/src/main/java/edu/illinois/cs/cogcomp/edison/annotators/SimpleGazetteerAnnotator.java:[72,23] error: cannot find symbol
[ERROR] 
[ERROR] could not parse error message:   symbol:   variable TREE_GAZETTEER
[ERROR] location: class ViewNames
[ERROR] /home/runner/illinois-cogcomp-nlp/edison/src/main/java/edu/illinois/cs/cogcomp/edison/annotators/SimpleGazetteerAnnotator.java:83: error: cannot find symbol
[ERROR] super(ViewNames.TREE_GAZETTEER, new String[] {ViewNames.SENTENCE, ViewNames.TOKENS});
[ERROR] ^
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :illinois-edison

**Comment by cowchipkid** [comment]
I had forgotten to check in ViewNames. It is checked in now, and appears to have been added to this existing PR, which surprised me a bit. Can I re-run semaphore on this?

**Comment by mssammon** [comment]
@cowchipkid are you able to click on the details link next to the "sempahoreci the build failed..." message above?

**Comment by cowchipkid** [comment]
No, I get a 404 when I click that link.

**Comment by cowchipkid** [comment]
I got it to run, there is still a unit test failure. Looking into it.

File: core-utilities/src/main/java/edu/illinois/cs/cogcomp/core/datastructures/ViewNames.java [hunk 1]
```diff
@@ -28,6 +28,9 @@
     public static final String LEMMA = "LEMMA";
     public static final String POS = "POS";
 
+    /** the tree gazetteer produced by the SimpleGazetteerAnnotator. */
+    public static final String TREE_GAZETTEER = "TREE_GAZETTEER";
```
Comment (mssammon) [inline core-utilities/src/main/java/edu/illinois/cs/cogcomp/core/datastructures/ViewNames.java]: please also add a switch case for this view under 'getViewType()'

**Comment by mssammon** [comment]
In addition to the ViewNames fix, github is currently reporting a conflict.

**Pull request merged**